src/LineBot/Events/LineEventCollection.cs

```diff
@@ -12,13 +12,35 @@
 // License for the specific language governing permissions and limitations
 // under the License.
 
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
 using Newtonsoft.Json;
 
 namespace Line
 {
-    internal sealed class LineEventCollection
+    [JsonObject]
+    internal sealed class LineEventCollection : ILineEvents
     {
+        [JsonProperty("destination")]
+        string ILineEvents.Destination { get; set; }
+
         [JsonProperty("events")]
-        public LineEvent[] Events { get; set; }
+        List<ILineEvent> ILineEvents.Events { get; set; }
+
+        public IEnumerator<ILineEvent> GetEnumerator()
+        {
+            return GetEnumerator();
```

Comment:
This is why your tests are failing. You are getting a stack overflow here. This method is calling itself. Should be changed to : `public IEnumerator<ILineEvent> GetEnumerator() => Events.GetEnumerator();` You can use the same implementation for the `IEnumerator IEnumerable.GetEnumerator()` method.